### PR TITLE
Slight TG Headset Communication port

### PR DESCRIPTION
**Pull request by Aurorablade**
So..We already had reply via headset for like..faxes..this kinda make is a verb. And adds an option to speak from syndicate.

My brain is not smart enough right now to make it add other options to it. Its an admin verb. Open to suggestions and improvement.

...Fethas go sleep now yes?

### CODE REVIEW

File: code/modules/admin/verbs/randomverbs.dm [hunk 1]
```diff
@@ -141,6 +141,46 @@
...
+	if(!input)
+		message_admins("[key_name_admin(src)] decided not to answer [key_name_admin(H)]'s [sender] request.")
+		return
+
+	log_admin("[key_name(src)] replied to [key_name(H)]'s [sender] message with the message [input].")
+	message_admins("[key_name_admin(src)] replied to [key_name_admin(H)]'s [sender] message with: \"[input]\"")
+	to_chat(H, "You hear something crackle in your ears for a moment before a voice speaks.  \"Please stand by for a message from [sender == "Syndicate" ? "your benefactor" : "Central Command"].  Message as follows[sender == "Syndicate" ? ", agent." : ":"] <span class='bold'>[input].</span> Message ends.\"")
+
+
+
```
Comment (Aurorablade) [inline code/modules/admin/verbs/randomverbs.dm]: AHHHHHHHHH WHY DID I PUT SO MANY SPACES..FETHAS FIX IT

File: code/modules/admin/verbs/randomverbs.dm [hunk 2]
```diff
@@ -141,6 +141,46 @@
...
+	message_admins("[key_name_admin(src)] has started answering [key_name_admin(H)]'s [sender] request.")
+	var/input = input("Please enter a message to reply to [key_name(H)] via their headset.", "Outgoing message from [sender]", "") as text|null
+	if(!input)
+		message_admins("[key_name_admin(src)] decided not to answer [key_name_admin(H)]'s [sender] request.")
+		return
+
+	log_admin("[key_name(src)] replied to [key_name(H)]'s [sender] message with the message [input].")
+	message_admins("[key_name_admin(src)] replied to [key_name_admin(H)]'s [sender] message with: \"[input]\"")
+	to_chat(H, "You hear something crackle in your ears for a moment before a voice speaks.  \"Please stand by for a message from [sender == "Syndicate" ? "your benefactor" : "Central Command"].  Message as follows[sender == "Syndicate" ? ", agent." : ":"] <span class='bold'>[input].</span> Message ends.\"")
```
Comment (Citinited) [inline code/modules/admin/verbs/randomverbs.dm]: `To return the call, press 1. There is normally a charge for this service` 🎉

File: code/modules/admin/admin.dm [hunk 3]
```diff
@@ -61,7 +61,8 @@ var/global/nologevent = 0
 	body += "<a href='?src=[usr.UID()];priv_msg=\ref[M]'>PM</a> - "
 	body += "<a href='?_src_=holder;subtlemessage=\ref[M]'>SM</a> - "
 	body += "[admin_jump_link(M)]\] </b><br>"
-
+	if(ishuman(M) && M.mind)
+		body += "<a href='?_src_=holder;HeadsetMessage=\ref[M]'>HM</a>"
```
Comment (Crazylemon64) [inline code/modules/admin/admin.dm]: `M.UID()`

File: code/modules/admin/topic.dm [hunk 4]
```diff
@@ -1675,22 +1675,22 @@
 		if(!check_rights(R_ADMIN))
 			return
 
-		var/mob/living/carbon/human/H = locate(href_list["CentcommReply"])
-		if(!istype(H))
-			to_chat(usr, "This can only be used on instances of type /mob/living/carbon/human")
-			return
-		if(!istype(H.l_ear, /obj/item/radio/headset) && !istype(H.r_ear, /obj/item/radio/headset))
-			to_chat(usr, "The person you are trying to contact is not wearing a headset")
+		var/mob/M = locate(href_list["CentcommReply"])
```
Comment (Crazylemon64) [inline code/modules/admin/topic.dm]: `locateUID`

File: code/modules/admin/topic.dm [hunk 5]
```diff
@@ -1675,22 +1675,22 @@
...
+		usr.client.admin_headset_message(M, "Centcomm")
+
+	else if(href_list["SyndicateReply"])
+		if(!check_rights(R_ADMIN))
 			return
 
-		var/input = input(src.owner, "Please enter a message to reply to [key_name(H)] via [H.p_their()] headset.","Outgoing message from Centcomm", "")
-		if(!input)	return
+		var/mob/M = locate(href_list["SyndicateReply"])
```
Comment (Crazylemon64) [inline code/modules/admin/topic.dm]: `locateUID`

File: code/modules/admin/topic.dm [hunk 6]
```diff
@@ -1675,22 +1675,22 @@
...
 
-		to_chat(src.owner, "You sent [input] to [H] via a secure channel.")
-		log_admin("[key_name(src.owner)] replied to [key_name(H)]'s Centcomm message with the message [input].")
-		message_admins("[key_name_admin(src.owner)] replied to [key_name_admin(H)]'s Centcom message with: \"[input]\"")
-		to_chat(H, "You hear something crackle in your headset for a moment before a voice speaks.  \"Please stand by for a message from Central Command.  Message as follows. [input].  Message ends.\"")
+	else if(href_list["HeadsetMessage"])
+		if(!check_rights(R_ADMIN))
+			return
 
+		var/mob/M = locate(href_list["HeadsetMessage"])
```
Comment (Crazylemon64) [inline code/modules/admin/topic.dm]: `locateUID`

**Pull request merged**